portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -44,34 +67,30 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String content = request.getParameter("comment");
     String name = request.getParameter("name");
-    
+    String stringifiedKey = request.getParameter("comment-key");
+
     if (content != null && name != null && content.length() != 0) {
-      Comment comment = name.length() == 0 ? new Comment(content) : new Comment(name, content);
+      name = name.length() == 0 ? "anonymous" : name;
+      Date timestamp = new Date();
       Entity commentEntity = new Entity("Comment");
-      commentEntity.setProperty("name", comment.getNAME());
-      commentEntity.setProperty("content", comment.getCONTENT());
-      commentEntity.setProperty("numLikes", comment.getNumLikes());
-      commentEntity.setProperty("timestamp", comment.getTIMESTAMP());
-      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      commentEntity.setProperty("name", name);
+      commentEntity.setProperty("content", content);
+      commentEntity.setProperty("numLikes", 0);
+      commentEntity.setProperty("timestamp", timestamp);
       datastore.put(commentEntity);
-      commentsList.add(comment);
-    } else {
+    } else if (stringifiedKey != null) {
       // Patch for numLikes (executes when comment is liked).
-      int id = Integer.parseInt(request.getParameter("comment-id"));
-      int commentInd = getObjectIndex(id);
-      if (commentInd >= 0) {
-        commentsList.get(commentInd).addLike();
+      Key key = KeyFactory.stringToKey(stringifiedKey);
+      try {
+        Entity retrievedComment = datastore.get(key);
+        long numLikes = (long) retrievedComment.getProperty("numLikes");
+        numLikes++;
+        retrievedComment.setProperty("numLikes", numLikes);
+        datastore.put(retrievedComment);
+      } catch (EntityNotFoundException e) {
```

Comment:
This exception couldn't be thrown in the header because then it wouldn't override the method

Reply:
To prevent this if else logic, you can create two different servlets - one for adding comments and one for adding likes. So for example, we could have NewCommentServlet and AddLikeServlet. Then these doPost functions can be dedicated to one thing.

Reply:
+1 - good idea, that should make this much cleaner:)